test/unit/ajax.js

```diff
@@ -1129,6 +1129,32 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - data - x-www-form-urlencoded (gh-2658)", 1, function( assert ) {
+		return {
+			url: "bogus.html",
+			data: { devo: "A Beautiful World" },
+			type: "post",
+			beforeSend: function( _, s ) {
+				assert.strictEqual( s.data, "devo=A+Beautiful+World", "data is '+'-encoded" );
+				return false;
+			},
+			error: true
+		};
+	} );
+
+	ajaxTest( "jQuery.ajax() - data - URL data isn't form encoded (gh-2658)", 1, function( assert ) {
+		return {
+			url: "bogus.html",
+			data: { devo: "A Beautiful World" },
+			type: "get",
+			beforeSend: function( _, s ) {
+				assert.strictEqual( s.url, "bogus.html?devo=A%20Beautiful%20World", "data is NOT '+'-encoded" );
```

Comment:
OK, since we've gotten several complaints about `+`-encoding of URL strings in the past I am ok with leaving it as well. I will remove this URL test for GET leaving that behavior unspecified, but add a few more unit tests for the POST case including one with a non-default content-type.